src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -119,39 +126,48 @@ public static void ReportNonFatal(Exception exception)
             session.PostEvent(faultEvent);
         }
 
-        private static List<string> CaptureLogFiles()
+        private static List<string> CollectServiceHubLogFilePaths()
         {
-            var result = new List<string>();
-            CollectServiceHubLogFiles(result);
-            CaptureFilesInMemory(result);
-            return result;
-        }
+            var paths = new List<string>();
 
-        private static void CollectServiceHubLogFiles(List<string> paths)
-        {
             try
             {
                 var logPath = Path.Combine(Path.GetTempPath(), "servicehub", "logs");
                 if (!Directory.Exists(logPath))
                 {
-                    return;
+                    return paths;
                 }
 
                 // attach all log files that are modified less than 1 day before.
                 var now = DateTime.UtcNow;
                 var oneDay = TimeSpan.FromDays(1);
 
-                foreach (var file in Directory.EnumerateFiles(logPath, "*.log"))
+                foreach (var path in Directory.EnumerateFiles(logPath, "*.log"))
                 {
                     try
                     {
-                        var lastWrite = File.GetLastWriteTimeUtc(file);
+                        var name = Path.GetFileNameWithoutExtension(path);
+
+                        // TODO: https://github.com/dotnet/roslyn/issues/42582 
+                        // name our services more consistently to simplify filtering
+
+                        // filter logs that are not relevant to Roslyn investigation
+                        if (!name.Contains("-" + WellKnownServiceHubServices.NamePrefix) &&
+                            !name.Contains("-CodeLens") &&
+                            !name.Contains("-pythia") &&
+                            !name.Contains("-ManagedLanguage.IDE.RemoteHostClient") &&
+                            !name.Contains("-hub"))
+                        {
+                            continue;
+                        }
+
+                        var lastWrite = File.GetLastWriteTimeUtc(path);
                         if (now - lastWrite > oneDay)
                         {
                             continue;
                         }
 
-                        paths.Add(file);
+                        paths.Add(path);
```

Comment:
Did we also want to limit this at a number like say 10 or 20?

Reply:
I wouldn't think so. If we do that we might not get the right logs.